### ISSUE TITLE
S3400: Don't raise issue for virtual methods

### ISSUE THREAD

**Issue by GeertvanHorrik** [SonarSource/sonar-csharp#1543]
### Description

Please provide a succinct description of your issue.

### Repro steps

Please provide the steps required to reproduce the problem

```
        protected virtual bool ShouldIncludeCollapsedRows()
        {
            return false;
        }

        protected virtual bool ShouldIncludeHiddenColumns()
        {
            return false;
        }

        protected virtual bool ShouldDiscardFilters()
        {
            return false;
        }
```

Virtual members are used to allow logic differentiation (we could discuss whether these should be properties, but that's out of scope for now).

Then getting this code smell:

```
Remove this method and declare a constant for this value.
```

### Expected behavior

Since virtual members can be overridden, you cannot replace them with constant values.

### Actual behavior

Raises a flag that the method should be replaced by a constant value.

### Known workarounds

N/A.

### Related information

* SonarC# Version
* Visual Studio Version
* If running through the Scanner for MSBuild, its version

**Comment by Evangelink** [comment]
Hi @GeertvanHorrik,

Thanks for the feedback (and sorry for this late answer). You are right that we should ignore virtual methods/

**Comment by andrei-epure-sonarsource** [comment]
Fixed in PR #1600
